tests/image_processing_tests/src/image_processor_service_test.cpp

```diff
@@ -45,28 +44,36 @@ TEST(ImageProcessingServiceTest, ImageProcessingServiceTest) {
                                                                                   nullptr,
                                                                                   nullptr);
 
-    SimpleWriteService writeService(nullptr, imageDataEventEmitter.getPublisher(), TEST_DATA, true);
+     kpsr::vision_ocv::FileImageStreamingService fileImageStreamingService(nullptr,
+                                                                          imageDataEventEmitter.getPublisher(),
+                                                                          TEST_DATA,
+                                                                          true,
+                                                                          480,
+                                                                          320,
+                                                                          "body",
+                                                                          "FileImageStreamingService");
+
 
-    kpsr_tutorial::ImageProcessingService imageProcessingServiceService(nullptr, imageDataEventEmitter.getSubscriber(), resultDataEventEmitter.getPublisher());
+    kpsr_tutorial::ImageProcessingService imageProcessingServiceService(nullptr, 0, imageDataEventEmitter.getSubscriber(), resultDataEventEmitter.getPublisher());
 
     kpsr::mem::TestCacheListener<kpsr_tutorial::ResultData> testListener(0);
 
     resultDataEventEmitter.getSubscriber()->registerListener("Test", testListener.cacheListenerFunction);
 
     spdlog::info("Starting services.");
-    writeService.startup();
+    fileImageStreamingService.startup();
     imageProcessingServiceService.startup();
 
     // Publish some integers.
     for (int i = 0; i < 1; i++) {
         spdlog::info("Executing services.");
-        writeService.runOnce();
+        fileImageStreamingService.runOnce();
         imageProcessingServiceService.runOnce();
 
         ASSERT_TRUE(testListener.getLastReceivedEvent()->ok);
     }
 
-    writeService.shutdown();
-    writeService.shutdown();
+    imageProcessingServiceService.shutdown();
+    fileImageStreamingService.shutdown();
```

Comment:
Although it is not necessary to remove the listener from the event emitter, it is good practice to add that line.